src/libstd/sync/mod.rs

```diff
@@ -17,17 +17,28 @@
 
 #![stable(feature = "rust1", since = "1.0.0")]
 
+#[stable(feature = "rust1", since = "1.0.0")]
```

Comment:
I'm a little wary about requiring stability annotations for reexports, I'm not sure we've hammered out a good set of semantics for them to follow. For example:

* What does it mean to have a stable reexport of an unstable item?
* What does it mean to have an unstable reexport of a stable item?

I'd expect both cases to be unstable, but I don't think anything is taking this into account today?